### PR TITLE
[script.module.inputstreamhelper] 0.4.2

### PR DESCRIPTION
**Pull request by mediaminister**
### Description

This release includes urgent fixes for two problems that unfortunately emerged after the 0.4.1 release:
- Fix unresponsive Kodi when opening add-on information pane
- Revert a commit that broke Chrome OS extracting for some platforms

This release also contains a new InputStream Helper information settings page and some small improvements regarding add-on structure and performance.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by dagwieers** [comment]
I approve of this new release.

File: script.module.inputstreamhelper/default.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+''' This is the actual InputStream Helper API script entry point '''
+
+from __future__ import absolute_import, division, unicode_literals
+import os
+import sys
```
Comment (enen92) [inline script.module.inputstreamhelper/default.py]: Why is this needed?

Comment (enen92) [reply script.module.inputstreamhelper/default.py]: I mean the sys.path.append

**Comment by Lunatixz** [comment]
LGTM

File: script.module.inputstreamhelper/default.py (hunk 1 above)
Comment (mediaminister) [reply script.module.inputstreamhelper/default.py]: Because default.py (**xbmc.python.script** extension point) needs access to  to functions of the **xbmc.python.module** extension point` but is not in the same directory.
In 0.4.1 we imported script.module.inputstreamhelper into itself to solve this problem, but that caused the Kodi add-on information pane to freeze.

Comment (enen92) [reply script.module.inputstreamhelper/default.py]: Hum, you guys are likely doing something wrong then. There's no need to append to sys.path inside the same add-on. I'd recommend to take a deeper look in the future to see which imports caused the circular dependency in the first place. For now I'm fine with it.

Comment (dagwieers) [reply script.module.inputstreamhelper/default.py]: Or we have been bitten by an issue I have experienced myself various times before, which causes a just installed script.module to not be added to PYTHONPATH. We have seen this for different modules and add-ons, so I doubt it is an issue with one specific module or add-on.

Possibly related to Kodi using the repository metadata and not the package metadata (as reported by @mediaminister).

Comment (basrieter) [reply script.module.inputstreamhelper/default.py]: @dagwieers I have also noticed this. Perhaps report it at https://github.com/xbmc/xbmc/issues/

Comment (dagwieers) [reply script.module.inputstreamhelper/default.py]: @basrieter Reported issues get closed without debug log, so I don't bother if it's not easily reproduced. Feel free to report it if you can.

Comment (basrieter) [reply script.module.inputstreamhelper/default.py]: Well, you cannot expect devs to look into stuff without the proper information. Just provide them the info and people will look into it.

Comment (dagwieers) [reply script.module.inputstreamhelper/default.py]: It depends on how you get about with reported issues. But I am not expecting anything from volunteers, as a volunteer. But in this case I didn't bother. And you didn't either :wink:
